In [39]:
import pandas as pd

# Sales Analysis

In [40]:
df = pd.read_csv("Superstore.csv", encoding="latin-1")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [41]:
df.Sales.sum()

np.float64(2297200.8603)

In [42]:
g = df.groupby("Region")["Sales"]

for region, region_df in g:
    print(region)
    print(region_df)

Central
14       68.8100
15        2.5440
16      665.8800
21       19.4600
22       60.3400
          ...   
9961     65.5840
9962    383.4656
9972     99.5680
9982     35.5600
9983     97.9800
Name: Sales, Length: 2323, dtype: float64
East
23        71.372
27      3083.430
28         9.618
29       124.200
30         3.264
          ...   
9968     735.980
9969      22.750
9981      16.520
9984      31.500
9985      55.600
Name: Sales, Length: 2848, dtype: float64
South
0       261.9600
1       731.9400
3       957.5775
4        22.3680
12       15.5520
          ...   
9971    140.7500
9980     85.9800
9987     79.9900
9988    206.1000
9989     25.2480
Name: Sales, Length: 1620, dtype: float64
West
2        14.620
5        48.860
6         7.280
7       907.152
8        18.504
         ...   
9986     36.240
9990     91.960
9991    258.576
9992     29.600
9993    243.160
Name: Sales, Length: 3203, dtype: float64


In [43]:
g.sum().sort_values(ascending=True)

Region
South      391721.9050
Central    501239.8908
East       678781.2400
West       725457.8245
Name: Sales, dtype: float64

The highest number of total sales took place in the West region which is 725457.8245 and least number of total sales took place in the South

In [44]:
g1 = df.groupby("Category")["Sales"]
g1.sum().sort_values(ascending=True)

Category
Office Supplies    719047.0320
Furniture          741999.7953
Technology         836154.0330
Name: Sales, dtype: float64

So the highest number of sales occured in the "Technology" category which were 836154.0330 and least number of sales occured in the "Office Supplies" category

In [45]:
print(df.Category.unique())

g2 = df[df["Category"] == "Furniture"].groupby("Sub-Category")["Sales"]
print(g2.sum().sort_values(ascending=True))

print("\n")

g3 = df[df["Category"] == "Office Supplies"].groupby("Sub-Category")["Sales"]
print(g3.sum().sort_values(ascending=True))

print("\n")

g4 = df[df["Category"] == "Technology"].groupby("Sub-Category")["Sales"]
print(g4.sum().sort_values(ascending=True))

<StringArray>
['Furniture', 'Office Supplies', 'Technology']
Length: 3, dtype: str
Sub-Category
Furnishings     91705.1640
Bookcases      114879.9963
Tables         206965.5320
Chairs         328449.1030
Name: Sales, dtype: float64


Sub-Category
Fasteners       3024.280
Labels         12486.312
Envelopes      16476.402
Art            27118.792
Supplies       46673.538
Paper          78479.206
Appliances    107532.161
Binders       203412.733
Storage       223843.608
Name: Sales, dtype: float64


Sub-Category
Copiers        149528.030
Accessories    167380.318
Machines       189238.631
Phones         330007.054
Name: Sales, dtype: float64


In the,

-> Furniture Category, the highest number of total sales were made by "Chairs" which is 328449.1030

-> Office Supplies Category, the highest number of total sales were made by "Storage" which is 223843.608

-> Technology Category, the highest number of total sales were made by "Phones" which is 330007.054

In [46]:
df["Order Date"] = pd.to_datetime(df["Order Date"])
df["Year"] = df["Order Date"].dt.year
df["Month"] = df["Order Date"].dt.month

# Profit Analysis

In [ ]:
df["Profit"].sum()

np.float64(442528.3074)

Total profit made is 286397.0217

In [80]:
grp1 = df.groupby("Category")["Profit"]
grp1.sum().sort_values(ascending=True)

Category
Furniture           18451.2728
Office Supplies    122490.8008
Technology         145454.9481
Name: Profit, dtype: float64

The higest profit is made by the "Technology" Category which is 145454.9481 and least proft was made by "Furniture" Category which is 18451.2728

In [72]:
profit_region = df[df["Region"] == "Central"].groupby("Profit")

for i, i_df in profit_region:
    print(i)
    print(i_df)

-3701.8928
      Row ID        Order ID Order Date  Ship Date       Ship Mode  \
9774    9775  CA-2014-169019 2014-07-26  7/30/2014  Standard Class   

     Customer ID Customer Name   Segment        Country         City  ...  \
9774    LF-17185   Luke Foster  Consumer  United States  San Antonio  ...   

           Product ID         Category Sub-Category  \
9774  OFF-BI-10004995  Office Supplies      Binders   

                                   Product Name     Sales Quantity Discount  \
9774  GBC DocuBind P400 Electric Binding System  2177.584        8      0.8   

         Profit  Year  Month  
9774 -3701.8928  2014      7  

[1 rows x 23 columns]
-2929.4845
      Row ID        Order ID Order Date   Ship Date       Ship Mode  \
4991    4992  US-2017-122714 2017-12-07  12/13/2017  Standard Class   

     Customer ID  Customer Name    Segment        Country     City  ...  \
4991    HG-14965  Henry Goldwyn  Corporate  United States  Chicago  ...   

           Product ID         Cat

In [84]:
discount_profit = df.groupby("Discount")["Profit"]
discount_profit.mean().sort_values(ascending=True)

Discount
0.50   -310.703456
0.45   -226.646464
0.40   -111.927429
0.80   -101.796797
0.70    -95.874060
0.32    -88.560656
0.30    -45.679636
0.60    -43.077212
0.20     24.702572
0.15     27.288298
0.00     66.900292
0.10     96.055074
Name: Profit, dtype: float64

According to the data a discount of 50% account for the highest loss

We start losing from a discount of 40%

In [89]:
consumer_sales = df.groupby("Customer Name")["Sales"]
consumer_sales.max().sort_values(ascending=False).head(10)

Customer Name
Sean Miller           22638.480
Tamara Chand          17499.950
Raymond Buch          13999.960
Tom Ashbrook          11199.968
Hunter Lopez          10499.970
Adrian Barton          9892.740
Sanjit Chand           9449.950
Bill Shonely           9099.930
Sanjit Engle           8749.950
Christopher Conant     8399.976
Name: Sales, dtype: float64

Top 10 consumers
1. Sean Miller
2. Tamara Chand  
3. Raymond Buch 
4. Tom Ashbrook 
5. Hunter Lopez
6. Adrian Barton          
7. Sanjit Chand           
8. Bill Shonely           
9. Sanjit Engle           
10. Christopher Conant     


In [92]:
region_profit = df.groupby("Region")["Profit"]
region_profit.sum()

Region
Central     39706.3625
East        91522.7800
South       46749.4303
West       108418.4489
Name: Profit, dtype: float64

The highest number of profit in sales was made in the West region with a profit of 108418.4489. 

The least number of profit in sales was made in the Central region with a profit of 39706.3625. 

In [97]:
product_analysis = df.groupby("Product Name")["Profit"]
product_analysis.sum().sort_values(ascending=False)

Product Name
Canon imageCLASS 2200 Advanced Copier                                          25199.9280
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind     7753.0390
Hewlett Packard LaserJet 3310 Copier                                            6983.8836
Canon PC1060 Personal Laser Copier                                              4570.9347
HP Designjet T520 Inkjet Large Format Printer - 24" Color                       4094.9766
                                                                                  ...    
Bush Advantage Collection Racetrack Conference Table                           -1934.3976
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases                       -2876.1156
Cubify CubeX 3D Printer Triple Head Print                                      -3839.9904
Lexmark MX611dhe Monochrome Laser Printer                                      -4589.9730
Cubify CubeX 3D Printer Double Head Print                                      -8879.97

The highest profit is made by Canon imageCLASS 2200 Advanced Copier  which is 25199.9280.

The highest loss is made by Cubify CubeX 3D Printer Double Head Print which is -8879.9704

In [100]:
df[["Sales", "Profit", "Discount"]].corr()

,Sales,Profit,Discount
Sales,1.000000,0.479064,-0.028190
Profit,0.479064,1.000000,-0.219487
Discount,-0.028190,-0.219487,1.000000


-> Highest sales then highest profits usually

-> More discount then less profits are made